fluss-common/src/main/java/org/apache/fluss/record/FileLogProjection.java

```diff
@@ -434,14 +398,130 @@ ByteBuffer getLogHeaderBuffer() {
+        BitSet nodesProjection = new BitSet(totalFieldNodes);
+        int totalBuffers = 0;
+        for (int i = 0; i < totalFieldNodes; i++) {
+            Field fieldNode = flattenedFields.get(i).f0;
+            boolean selected = flattenedFields.get(i).f1;
+            nodesProjection.set(i, selected);
+            bufferLayoutCount[i] = TypeLayout.getTypeBufferCount(fieldNode.getType());
+            totalBuffers += bufferLayoutCount[i];
+        }
+        BitSet buffersProjection = new BitSet(totalBuffers);
+        int bufferIndex = 0;
+        for (int i = 0; i < totalFieldNodes; i++) {
+            if (nodesProjection.get(i)) {
+                buffersProjection.set(bufferIndex, bufferIndex + bufferLayoutCount[i]);
+            }
+            bufferIndex += bufferLayoutCount[i];
+        }
+
+        Schema projectedArrowSchema =
+                ArrowUtils.toArrowSchema(rowType.project(selectedFieldPositions));
+        ArrowBodyCompression bodyCompression =
+                CompressionUtil.createBodyCompression(compressionInfo.createCompressionCodec());
+        int metadataLength =
+                ArrowUtils.estimateArrowMetadataLength(projectedArrowSchema, bodyCompression);
+        currentProjection =
+                new ProjectionInfo(
+                        nodesProjection,
+                        buffersProjection,
+                        bufferIndex,
+                        rowType,
+                        metadataLength,
+                        bodyCompression,
+                        selectedFieldPositions,
+                        selectedFieldIds);
+        projectionsCache.put(projectionKey, currentProjection);
+    }
+
+    int[] checkProjection(org.apache.fluss.metadata.Schema schema, int[] projectedFields) {
+        Map<Integer, Integer> columnIdPositions = new HashMap<>();
+        List<Integer> columnIds = schema.getColumnIds();
+        for (int i = 0; i < columnIds.size(); i++) {
+            columnIdPositions.put(columnIds.get(i), i);
+        }
+
+        int prev = -1;
+        int[] selectedFieldPositions = new int[projectedFields.length];
+        for (int i = 0; i < projectedFields.length; i++) {
+            int fieldId = projectedFields[i];
+            Integer position = columnIdPositions.get(fieldId);
+            if (position == null) {
+                throw new InvalidColumnProjectionException(
+                        String.format(
+                                "Projected field id %s is not contains in %s", fieldId, columnIds));
+            }
+
+            selectedFieldPositions[i] = position;
+            if (position < prev) {
+                throw new InvalidColumnProjectionException(
+                        "The projection indexes should be in field order, but is "
+                                + Arrays.toString(projectedFields));
+            }
+
+            prev = position;
+        }
+        return selectedFieldPositions;
```

Comment:
We can remove this for now, as it may affect projection performance, and we don't need the remapping, as we only support add column add end. We can rename the method into `toSelectedFieldPositions()`.